compiler/rustc_abi/src/layout.rs

```diff
@@ -39,7 +39,7 @@ enum NicheBias {
     End,
 }
 
-#[derive(Copy, Clone, Debug)]
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
```

Comment:
I think rust-analyzer's derives of these are probably unnecessary but I'm not gonna fight that war today.

Reply:
I think they're needed because we store these into a `salsa` database.

Reply:
Huh, interesting.